### PR TITLE
[OneCollector] Integration tests prep

### PR DESCRIPTION
**Pull request by CodeBlanch**
## Changes

* Adds a new workflow for running OneCollector integration tests. Those tests are being added on #1347

## Details

The goal is simple: We have a secret defined `OTEL_ONECOLLECTOR_INSTRUMENTATION_KEY` to set as an environment variable of the same name when running the tests.

Turns out this is not super easy to do!

A workflow defined as...

```yaml
on:
  pull_request:
```

...when opened from a fork does NOT see secrets.

There is a special trigger for this: [pull_request_target](https://github.blog/2020-08-03-github-actions-improvements-for-fork-and-pull-request-workflows/#improvements-for-public-repository-forks).

But the problem with `pull_request_target` is, it runs on the targert repo, not the fork. That means the integration tests won't run with the code being PRed (which is what we want).

The reason for this is, there is an attack vector: [pwn requests](https://securitylab.github.com/research/github-actions-preventing-pwn-requests/).

I did a bunch of reading on this and decided to try the [approach iterative took](https://iterative.ai/blog/testing-external-contributions-using-github-actions-secrets). Their approach is to require a human to approve these runs presumably after they have verified the code which will be run is safe. I think the manual approval approach is fine for these integration tests. I set up an environment called "external" which any approver or maintainer group member can approve.

So why is this PR happening? I can't get this `pull_request_target` workflow to run as part of #1347. It seems it needs to be part of the repo first. **Assuming no one objects, what I would like to do is merge this PR so I can try it out in action on #1347.** It is a bit of an experiment and might be reverted. This is either crazy or brilliant 🤣

### CODE REVIEW

**Review by alanwest — APPROVED** [review]
LGTM. Discussed with @CodeBlanch, he said he'll test this as soon as it's merged and validate it works as expected.

**Comment by utpilla** [comment]
@CodeBlanch Could you test the workflow on you forked repo instead? You should be able to store secrets on your forked repo and trigger the new workflow manually to test it.

**Review by Kielek — APPROVED** [review]

**Review by utpilla — CHANGES_REQUESTED** [review]
https://github.com/open-telemetry/opentelemetry-dotnet-contrib/pull/1348#issuecomment-1718755555

**Review by utpilla — APPROVED** [review]

**Comment by CodeBlanch** [comment]
Just as an FYI we performed some testing of this new workflow before merging. Using `CodeBlanch/opentelemetry-dotnet-contrib` as the "main repo" and `alanwest/opentelemetry-dotnet-contrib` as the "fork"...

* The "main repo" was seeded with the "external" environment (requiring approval) & the secret
* This PR was put into `main-test` branch on the "main repo"
* We opened 3 PRs from the "fork" doing (more or less) what #1347 is doing:
  * https://github.com/CodeBlanch/opentelemetry-dotnet-contrib/pull/12
  * https://github.com/CodeBlanch/opentelemetry-dotnet-contrib/pull/13
  * https://github.com/CodeBlanch/opentelemetry-dotnet-contrib/pull/14

@alanwest @utpilla and myself each verified the workflow did not run until it was approved, and it was able to see the secrets for its tests.

**Pull request merged**
